packages/experiments/src/utilities/VariableProcessing.ts

```diff
@@ -1,10 +1,18 @@
-import { merge } from '../Utilities';
+export type IComponentStyleVariables<TComponentViewProps, TComponentStyleVariablesTypes> =
+  | TComponentStyleVariablesTypes
+  | ((props: TComponentViewProps) => TComponentStyleVariablesTypes)
+  | undefined;
 
-export type IProcessedVariables<T> = { [P in keyof T]-?: IProcessedVariables<T[P]> };
+export function resolveStyleVariables<
+  TViewProps,
+  TStyleVariablesTypes,
+  TStyleVariables extends IComponentStyleVariables<TViewProps, TStyleVariablesTypes>
+>(props: TViewProps, componentStyleVariables: TStyleVariablesTypes, userStyleVariables: TStyleVariables): TStyleVariablesTypes {
```

Comment:
suggestion: this can be simplified into basically one line with less repeat:

```tsx
const resolveStyleVariables = (props, ...variableArray) => Object.assign(
  {},
  ...variableArray.map(
    variables => typeof variables === 'function' ?  variables(props) : variables
  )
);
```